keras/models.py

```diff
@@ -73,10 +73,10 @@ def weighted(y_true, y_pred, weights, mask=None):
         weighted = filtered_weights * obj_output
         if mask is None:
             # Instead of calling mean() here, we divide by the sum of filtered_weights.
-            return weighted.sum() / filtered_weights.sum()
+            return weighted.sum() / (filtered_weights.sum() + 1e-20)
```

Comment:
If we were to use a epsilon here... it should be abstracted as a globally configured epsilon. Also, I think this value is too low and would cause issues with GPUs (float32).

But anyway: we shouldn't. If the sum of the weights is zero, then the problem is ill-defined and a loss cannot be computed. We should throw an exception in that case.